src/hooks/use-load-data-pages.ts

```diff
@@ -36,6 +37,46 @@ export const isMissingLayoutData = (layout: PivotLayout, pageInfo: PageInfo): bo
   return qWidth < Math.min(DEFAULT_PAGE_SIZE, qSize.qcx) || qHeight < Math.min(DEFAULT_PAGE_SIZE, qSize.qcy);
 };
 
+/**
+ * Note: the state of the viewService is always from the last time the chart was rendered. So it is not
+ * guaranteed to work with a new layout, as its properties may no longer be valid.
+ */
+export const getFetchArea = (
+  qLastExpandedPos: EngineAPI.INxCellPosition | undefined,
+  viewService: ViewService,
+  qSize: EngineAPI.ISize,
+  pageInfo: PageInfo,
+) => {
+  const pageStartTop = pageInfo.page * pageInfo.rowsPerPage;
+  const pageEndTop = pageStartTop + pageInfo.rowsPerPage;
```

Comment:
if we sum start of the page with rows per page, then it should not be pageEndTop -> probably something like pageEnd(index?) should be good maybe (or any other suggestion)

also pageStartIndex instead of pageStartTop maybe?